src/isomorphic/classic/element/ReactElementValidator.js

```diff
@@ -143,7 +121,15 @@ function validateChildKeys(node, parentType) {
     for (var i = 0; i < node.length; i++) {
       var child = node[i];
       if (ReactElement.isValidElement(child)) {
-        validateExplicitKey(child, parentType);
+        const getMainExplicitKeyErrorMessage = getExplicitKeyErrorMessage.bind(
+          null,
+          parentType,
+        );
```

Comment:
I think the cleanest option is to just move it outside of the 'for' loop. I'll play around with some alternatives and maybe push another version to see, but I think it will be ugly.